org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java

```diff
@@ -528,11 +528,13 @@ protected XtextResource getState() {
 					throw e.getWrapped();
 				} finally {
 					try {
-						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) { 
+						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) {
+							boolean wasHadUpdates = hadUpdates;
```

Comment:
strange variable name.
how about setting hadUpdates in a finally-block to false after notifying the listeners? I did not see that it is mandatory to set it to false before.

```
try {
   ...
   if (hasUpdates) { notifyModelListeners(state); }
} finally {
   hadUpdates = false;
}
```

Reply:
i just kept the existing styles ;)